plugins/status/plugin.go

```diff
@@ -270,6 +289,14 @@ func (p *Plugin) oneShot(ctx context.Context) error {
 		}
 	}
 
+	if p.config.Plugin != nil {
```

Comment:
❓ This seems like a late stage for checking it -- we're already attempting to output logs (`oneShot`). I would think that this check is something we could do in `New`, too, unless we need this sort of late binding because the plugin used could have changed in the mean time. I'm genuinely unaware 😄

Reply:
i just copied the pattern from the decision logs plugin, but i see your point.  i suppose the plugin could change due to a reconfigure, but i could refactor it to set the proxy up as a property on plugin, and do the binding in both new and reconfigure, thereby avoiding the late re-binding each time.  thoughts?

Reply:
> i could refactor it to set the proxy up as a property on plugin, and do the binding in both new and reconfigure, thereby avoiding the late re-binding each time. thoughts?

This sounds reasonable to me. But I'm not the most well-versed in the plugins codebase... Anyone else having thoughts there?

Reply:
@gshively11's change is what I had in mind. The in-tree implementation has an opportunity to do whatever it wants w/ the status update before calling out to the plugin and returning. This is the same as how the decision log implementation works and I think it's been useful. In this case, e.g., users could have both plugin and console status logging enabled (which could be useful in debug scenarios.) If in the future, there is some cross-cutting behaviour that needs to be implemented in oneShot then this implementation will accommodate (similar to how we perform decision log masking _before_ calling out to decision log plugins...)